src/lib/mongoBackend/mongoQueryContext.cpp

```diff
@@ -38,6 +38,268 @@
 
 /* ****************************************************************************
 *
+* someContextElementNotFound -
+*
+* Returns true if some attribut with 'found' set to 'false' is found in the CER vector passed
+* as argument
+*
+*/
+bool someContextElementNotFound(ContextElementResponseVector& cerV)
+{
+  for (unsigned int ix = 0; ix < cerV.size(); ++ix)
+  {
+    for (unsigned int jx = 0; jx < cerV.get(ix)->contextElement.contextAttributeVector.size(); ++jx)
+    {
+      if (!cerV.get(ix)->contextElement.contextAttributeVector.get(jx)->found)
+      {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+
+/* ****************************************************************************
+*
+* searchCprForAttribute -
+*
+* Search the CPr, given the entity/attribute as argument. Actually, two CPrs can be returned
+* the "general" one at entity level or the "specific" one at attribute level
+*
+*/
+void searchCprForAttribute(EntityId& en, std::string attrName, ContextRegistrationResponseVector& crrV, std::string& perEntPa, std::string& perAttrPa)
+{
+  perEntPa  = "";
+  perAttrPa = "";
+  for (unsigned int ix = 0; ix < crrV.size(); ++ix)
+  {
+    ContextRegistrationResponse* crr = crrV.get(ix);
+    /* Is there a matching entity in the CRR? */
+    for (unsigned jx = 0; jx < crr->contextRegistration.entityIdVector.size(); ++jx)
+    {
+      EntityId* regEn = crr->contextRegistration.entityIdVector.get(jx);
+      if (regEn->id == en.id && (regEn->type == en.type || regEn->type == ""))
+      {
+        if (crr->contextRegistration.contextRegistrationAttributeVector.size() == 0)
+        {
+          /* CRR without attributes (we need to keep searching in other CRR) */
+          perEntPa = crr->contextRegistration.providingApplication.get();
+          break; /* jx */
+        }
+        else
```

Comment:
As the previous if breaks if entered, there is no reason for the 'else':

    if (crr->contextRegistration.contextRegistrationAttributeVector.size() == 0)
    {
        /* CRR without attributes (we need to keep searching in other CRR) */
       perEntPa = crr->contextRegistration.providingApplication.get();
       break; /* jx */
    }

    // Is there a matching entity or the abcense of attributes?
    for (unsigned kx = 0; kx < crr->contextRegistration.contextRegistrationAttributeVector.size(); ++kx)

This is another 'pre-condition'.
As always, this best practice saves you lines and intentations (adding to the readibility).